packages/web-components/fast-foundation/src/avatar/avatar.template.ts

```diff
@@ -9,19 +9,14 @@ export function avatarTemplate<T extends FASTAvatar>(
     options: AvatarOptions = {}
 ): ElementViewTemplate<T> {
     return html<T>`
-    <div
-        class="backplate"
-        part="backplate"
-    >
-        <a
-            class="link"
-            part="link"
-            href="${x => (x.link ? x.link : void 0)}"
-        >
-            <slot name="media">${options.media ?? ""}</slot>
-            <slot><slot>
-        </a>
-    </div>
-    <slot name="badge"></slot>
+        <div class="backplate" part="backplate">
+            <div class="media" part="media">
+                <slot name="media">${options.media ?? ""}</slot>
+            </div>
+            <div class="content" part="content">
+                <slot></slot>
+            </div>
+        </div>
+        <slot name="badge"></slot>
```

Comment:
This is adding a lot of DOM and I'm not sure I understand the reasoning around why. Is there a reason we're adding significantly more DOM when we're removing an entire layer here?

Reply:
The layer wa removed because we extracted the link functionality. The new wrappers are there because there is no way to target the individual containers to style the avatar. If someone wants to make a consistent avatar treatment they need to be able to style the component in general, not the slotted content for each instance. This follows the need for many of the "content" wrappers, as well as the consistency of having a slot wrapped by a class and part.

Reply:
> The layer wa removed because we extracted the link functionality. The new wrappers are there because there is no way to target the individual containers to style the avatar. If someone wants to make a consistent avatar treatment they need to be able to style the component in general, not the slotted content for each instance. This follows the need for many of the "content" wrappers, as well as the consistency of having a slot wrapped by a class and part.

I think this quote probably illustrates where we're coming at this from fairly different approaches. I don't actually think this is the minimum amount of DOM necessary to create a consistent avatar in general. Using the Fluent exploration I did previously, in an effort to align with their templates we were able to get a consistent avatar, with fairly significant configurable options, using the following DOM:

```ts
export const avatarTemplate = html<Avatar>`
    <template role="img">
        <span class="initials" part="initials"><slot name="initials">${x => x.initials}</slot></span>
        <slot>${when(x => !x.initials, 
            html`
                <svg width="1em" height="1em" viewBox="0 0 20 20" xmlns="http://www.w3.org/2000/svg" aria-hidden="true">
                    <path d="M10 2a4 4 0 100 8 4 4 0 000-8zM7 6a3 3 0 116 0 3 3 0 01-6 0zm-2 5a2 2 0 00-2 2c0 1.7.83 2.97 2.13 3.8A9.14 9.14 0 0010 18c1.85 0 3.58-.39 4.87-1.2A4.35 4.35 0 0017 13a2 2 0 00-2-2H5zm-1 2a1 1 0 011-1h10a1 1 0 011 1c0 1.3-.62 2.28-1.67 2.95A8.16 8.16 0 0110 17a8.16 8.16 0 01-4.33-1.05A3.36 3.36 0 014 13z"></path>
                </svg>
            `)}</slot>
        <slot name="badge"></slot>
    </template>
`;
```

Most of what is above is actually default slotted content. Without it, the template would be as thin as below.
```ts
export const avatarTemplate = html<Avatar>`
    <template role="img">
        <span class="initials" part="initials">
            <slot name="initials"></slot>
        </span>
        <slot></slot>
        <slot name="badge"></slot>
    </template>
`;
```

This results in what can be seen [here](https://chrisdholt.github.io/fluent-v9-example-wc/?path=/story/components-avatar--avatar).

I think this is where we probably need to get aligned on goals and priorities for templates. In many cases, I'm open to scenarios where we need a wrapper for things like badge, button, etc around the content to enable things like truncation. If that can't yet be done without a wrapping element, it's very easy for us to justify the purpose of that DOM. I don't believe though that additional wrapping DOM is necessary around slotted content in an effort to consistently style them, and I think in many cases we end up with a heavier than necessary DOM, which is a trade-off in performance.